test/common/upstream/health_checker_impl_test.cc

```diff
@@ -95,6 +95,18 @@ TEST(HealthCheckerFactoryTest, CreateGrpc) {
                              .get()));
 }
 
+class HealthCheckerTestBase {
+public:
+  std::shared_ptr<MockClusterMockPrioritySet> cluster_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  MockHealthCheckEventLogger* event_logger_{};
```

Comment:
std::unique_ptr<MockHealthCheckEventLogger> here, and use event_logger_.release() when passing ownership to another object. Also you can init-in-place rather than using a ctor:

```
std::unique_ptr<MockHealthCheckEventLogger> event_logger_{std::make_unique<MockHealthCheckEventLogger>*};
```

Reply:
Is it make sense to use a raw pointer?

Reply:
My suggestion here is to *not* use a raw pointer.

"Does it make sense to use a raw pointer": yes but modern C++ style tries to avoid this because the language constructs don't strongly indicate ownership. Instead we use unique_ptr and transfer ownership where needed via `.release()`.

It's probably worth reading through Google's style guide https://google.github.io/styleguide/cppguide.html , which Envoy follows with variations called out here: https://github.com/envoyproxy/envoy/blob/master/STYLE.md

There are still places where raw pointers make sense but they are increasingly rare.